### PR TITLE
Prepare alerts for mimir migration

### PR DESCRIPTION
**Pull request by QuentinBisson**
Before adding a new alerting rule into this repository you should consider creating an SLO rules instead.
SLO helps you both increase the quality of your monitoring and reduce the alert noise.

* How to create a SLO rule: https://github.com/giantswarm/sloth-rules#how-to-create-a-slo
* Documentation: https://intranet.giantswarm.io/docs/monitoring/slo-alerting/

---
This PR prepares GS alerts towards mimir migration. It mostly adds labels that were added via external-labels to all aggregations (cluster_id, cluster_type, customer, installation, pipeline, provider, region)

This PR does not fix the alerts using absent because the targets are on MC only (operators)

### Checklist

- [ ] Update CHANGELOG.md (not needed because this is already in changelog)
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

**Comment by QuentinBisson** [comment]
@hervenicol i'd love your help :)

**Review by QuantumEnigmaa — APPROVED** [review]

File: helm/prometheus-rules/templates/alerting-rules/cert-manager.rules.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
           If memory usage value is equal to memory limit value then it is likely the pod will be evicted.
           If no limits are set then the pod will burst.
           `}}
-      expr: sum by (pod, namespace, container) (container_memory_working_set_bytes{container=~"(cert-manager)"}) / sum by (pod, namespace, container) (kube_pod_container_resource_requests_memory_bytes{container=~"(cert-manager)"}) >= 0.85
+      expr: sum by (cluster_id, pod, namespace, container) (container_memory_working_set_bytes{container=~"(cert-manager|cert-manager-app-controller)"}) / sum by (cluster_id, pod, namespace, container) (kube_pod_container_resource_requests{resource="memory", unit="byte",container=~"(cert-manager|cert-manager-app-controller)"}) >= 0.85
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/cert-manager.rules.yml:20]: This fixes the cert-manager alert for newer versions

File: helm/prometheus-rules/templates/alerting-rules/certificate.all.rules.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ spec:
   groups:
   - name: certificate.all
     rules:
+    {{- if eq .Values.managementCluster.provider.flavor "vintage" }}
     - alert: KiamCertificateSecretWillExpireInLessThanTwoWeeks
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/certificate.all.rules.yml:14]: Kiam is vintage only

File: helm/prometheus-rules/templates/alerting-rules/dex.rules.yml [hunk 3]
```diff
@@ -29,7 +29,7 @@ spec:
       annotations:
         description: '{{`dex-operator failed to renew secret of {{ $labels.app_registration_name }} for {{ $labels.app_owner }} on provider {{ $labels.provider_type }}.`}}'
         opsrecipe: dex-operator/
-      expr: min by(app_registration_name, app_owner, app_namespace, provider_name, provider_type, installation) (aggregation:dex_operator_idp_secret_expiry_time{cluster_type="management_cluster"}) - time() < 60*60*12
+      expr: min by(app_registration_name, app_owner, app_namespace, provider_name, provider_type, cluster_id) (aggregation:dex_operator_idp_secret_expiry_time{cluster_type="management_cluster"}) - time() < 60*60*12
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/dex.rules.yml:32]: I think it causes no harm to keep the installation name, and maybe it was there by purpose.
I'd rather keep it and not risk breaking stuff on alerts we're not responsible for more than needed.
```suggestion
      expr: min by(app_registration_name, app_owner, app_namespace, provider_name, provider_type, installation, cluster_id) (aggregation:dex_operator_idp_secret_expiry_time{cluster_type="management_cluster"}) - time() < 60*60*12
```

File: helm/prometheus-rules/templates/alerting-rules/dns-operator-azure.rules.yml [hunk 4]
```diff
@@ -31,7 +31,7 @@ spec:
               {{`Error rate for {{ $labels.method }} is high. Check dns-operator-azure logs in installation/{{ $labels.installation }}.`}}
             opsrecipe: dns-operator-azure/
           expr: |-
-            sum by (method,installation) (rate(dns_operator_azure_api_request_errors_total[5m])) > 0
+            sum by (cluster_id, method) (rate(dns_operator_azure_api_request_errors_total[5m])) > 0
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/dns-operator-azure.rules.yml:34]: Same here: I wouldn't "clean up" if not needed.
```suggestion
            sum by (cluster_id, method, installation) (rate(dns_operator_azure_api_request_errors_total[5m])) > 0
```

File: helm/prometheus-rules/templates/alerting-rules/fluentbit.rules.yml [hunk 5]
```diff
@@ -46,7 +46,7 @@ spec:
       annotations:
         description: '{{`Fluentbit is down on node ({{ $labels.node }}).`}}'
         opsrecipe: fluentbit-down/
-      expr: sum(up{app="fluent-logshipping-app"}) by (app, cluster_id, cluster_type, installation, job, namespace, provider, node) == 0
+      expr: sum(up{app="fluent-logshipping-app"}) by (app, cluster_id, job, namespace, node) == 0
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/fluentbit.rules.yml:49]: All right, this one is ours to manage and maintain, so let's clean it up! :rofl: :+1:

File: helm/prometheus-rules/templates/alerting-rules/cert-manager.rules.yml (hunk 1 above)
Comment (hervenicol) [reply helm/prometheus-rules/templates/alerting-rules/cert-manager.rules.yml:20]: But do we run this new version all around? How can we be assured of that?

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/cert-manager.rules.yml:20]: This is the version fix "cert-manager|cert-manager-app-controller"

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/cert-manager.rules.yml:20]: This kube_pod_container_resource_requests was added in ksm 2.0.0

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/cert-manager.rules.yml:20]: And according to this we do not run ksm older than 2.6 https://giantswarm.grafana.net/explore?schemaVersion=1&panes=%7B%229rq%22:%7B%22datasource%22:%22000000006%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22aggregation:container:images%7Bimage%3D~%5C%22.%2Akube-state-metrics.%2Av1.%2A%5C%22%7D%22,%22range%22:true,%22instant%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D,%22editorMode%22:%22code%22,%22legendFormat%22:%22__auto%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1

**Comment by QuentinBisson** [comment]
Now, apart from fixing the tests, we are left with prometheus-agent and ksm absent alerts to fix and then we should be fine

File: helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml [hunk 6]
```diff
@@ -59,7 +59,6 @@ spec:
         cluster_status_deleting: "true"
         team: phoenix
         topic: status
-    {{- if eq .Values.managementCluster.provider.kind "aws" }}
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml:62]: wow, just noticed that the only vintage provider we support is now aws! :exploding_head:

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml:62]: I did a lot of work around that last week https://github.com/giantswarm/prometheus-rules/blob/master/CHANGELOG.md#removed

File: helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml [hunk 7]
```diff
@@ -27,6 +27,7 @@ spec:
         severity: notify
         team: {{ include "providerTeam" . }}
         topic: kubernetes
+    {{- if eq .Values.managementCluster.provider.kind "aws" }}
     - alert: AWSWorkloadClusterNodeTooManyAutoTermination
```
[the remark this answers is not in the document]
Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml:31]: Maybe they should but they are based on this metric which is vintage only
`increase(aws_operator_unhealthy_node_termination_count[60m])`

**Comment by QuentinBisson** [comment]
Thank you for the really deep review :) Let's release tomorrow morning

**Pull request merged**